### PR TITLE
Hide comments with low rating; #380

### PR DESCRIPTION
**Pull request by kiritofeng**

### CODE REVIEW

File: templates/comments/list.html [hunk 1]
```diff
@@ -8,7 +8,11 @@ <h2><i style="padding-right: 0.3em" class="fa fa-comments"></i>{{ _('Comments')
                 <li id="comment-{{ node.id }}" data-revision="{{ node.revisions - 1 }}"
                     data-max-revision="{{ node.revisions - 1 }}"
                     data-revision-ajax="{{ url('comment_revision_ajax', node.id) }}" class="comment">
-                    <div class="comment-display">
+                    {% if node.score > vote_hide_threshold %}
+                        <div class="comment-display">
+                    {% else %}
+                        <div class="comment-display bad-comment">
```
Comment (quantum5) [inline templates/comments/list.html]: Put the `if` inline. This messes up the indentation completely.

File: templates/comments/media-js.html [hunk 2]
```diff
@@ -49,6 +49,12 @@
 
             window.show_revision = function (comment_id, offset) {
                 var $comment = $("#comment-" + comment_id);
+
+                // If .real-comment-body is hidden, then this is a bad comment that the user has not clicked
+                // Thus the revision retrieval should do nothing
+                if($comment.find('.real-comment-body').css('display') == 'none')
```
Comment (quantum5) [inline templates/comments/media-js.html]: Space before `(`, and you probably want `.is(':visible')`.

File: templates/comments/media-js.html [hunk 3]
```diff
@@ -188,6 +194,12 @@
             });
 
             $('img.unveil').unveil(200);
+
+            window.comment_show_content = function (comment_id) {
+                var $comment = $("#comment-" + comment_id);
+                $comment.find('.bad-comment-body').css({display: 'none'});
```
Comment (quantum5) [inline templates/comments/media-js.html]: `.css('display', 'none')` like a normal person. But really, you want `.hide()`.

File: templates/comments/media-js.html [hunk 4]
```diff
@@ -188,6 +194,12 @@
...
+                $comment.find('.real-comment-body').css({display: 'inherit'});
```
Comment (quantum5) [inline templates/comments/media-js.html]: `.css('display', 'none')` like a normal person. But really, you want `.show()`.

File: templates/comments/list.html [hunk 5]
```diff
@@ -8,7 +8,7 @@ <h2><i style="padding-right: 0.3em" class="fa fa-comments"></i>{{ _('Comments')
                 <li id="comment-{{ node.id }}" data-revision="{{ node.revisions - 1 }}"
                     data-max-revision="{{ node.revisions - 1 }}"
                     data-revision-ajax="{{ url('comment_revision_ajax', node.id) }}" class="comment">
-                    <div class="comment-display">
+                    {% if node.score > vote_hide_threshold %}<div class="comment-display">{% else %}<div class="comment-display bad-comment">{% endif %}
```
Comment (Xyene) [inline templates/comments/list.html]: I think @quantum5 meant to put the if inside the `class="..."`.

File: templates/comments/media-css.html [hunk 6]
```diff
@@ -91,5 +91,19 @@
         .comments.top-level-comments {
             margin-bottom: -3px;
         }
+
+        .bad-comment {
```
Comment (Xyene) [inline templates/comments/media-css.html]: You renamed this to `bad-comment-body` in the template, is this still correct here?

File: templates/comments/media-js.html [hunk 7]
```diff
@@ -49,6 +49,12 @@
 
             window.show_revision = function (comment_id, offset) {
                 var $comment = $("#comment-" + comment_id);
+
+                // If .real-comment-body is hidden, then this is a bad comment that the user has not clicked
+                // Thus the revision retrieval should do nothing
+                if(! $comment.find('.real-comment-body').is(':visible'))
```
Comment (Xyene) [inline templates/comments/media-js.html]: That's not where the space goes.

File: templates/comments/media-css.html (hunk 6 above)
Comment (kiritofeng) [reply templates/comments/media-css.html]: Refers to https://github.com/DMOJ/site/blob/hiding_comments/templates/comments/list.html#L11

File: templates/comments/list.html [hunk 8]
```diff
@@ -94,7 +94,19 @@ <h2><i style="padding-right: 0.3em" class="fa fa-comments"></i>{{ _('Comments')
...
-                                <div class="comment-body">
+                                    {% if node.score > vote_hide_threshold %}
+                                        <div class="comment-body real-comment-body">
+                                    {% else %}
+                                        <div class="comment-body bad-comment-body">
+                                            <p>
+                                                {% trans id=node.id %}
+                                                    This comment is hidden due to too much negative feedback.
+                                                    Click <a href="javascript:comment_show_content({{ id }})">here</a> to view it.
+                                                {% endtrans%}
```
Comment (quantum5) [inline templates/comments/list.html]: Space before `%`.

File: templates/comments/list.html [hunk 9]
```diff
@@ -94,7 +94,19 @@ <h2><i style="padding-right: 0.3em" class="fa fa-comments"></i>{{ _('Comments')
                             <div class="content content-description">
                                 <div class="comment-title"{% if not node.title or node.title.startswith('Re:') %}
                                      style="display: none"{% endif %}>{{ node.title }}</div>
-                                <div class="comment-body">
+                                    {% if node.score > vote_hide_threshold %}
```
Comment (quantum5) [inline templates/comments/list.html]: How about you show `real-comment-body` unconditionally, and add `style="display: none"` as an inline `if`? Then, conditionally display `bad-comment-body`. This way, the indentation actually lines up properly.

Alternatively, use CSS to hide the `real-comment-body`:

```css
.bad-comment-body + .real-comment-body {
    display: none;
}
```

File: templates/comments/media-js.html [hunk 10]
```diff
@@ -188,6 +194,12 @@
             });
 
             $('img.unveil').unveil(200);
+
+            window.comment_show_content = function (comment_id) {
+                var $comment = $("#comment-" + comment_id);
```
Comment (quantum5) [inline templates/comments/media-js.html]: Use single quotes.

**Comment by kiritofeng** [comment]
Turns out naming it `real-comment-body` interferes with history retrieval, so I'm going to stick with `comment-body`

File: templates/comments/list.html [hunk 11]
```diff
@@ -94,9 +94,19 @@ <h2><i style="padding-right: 0.3em" class="fa fa-comments"></i>{{ _('Comments')
                             <div class="content content-description">
                                 <div class="comment-title"{% if not node.title or node.title.startswith('Re:') %}
                                      style="display: none"{% endif %}>{{ node.title }}</div>
-                                <div class="comment-body">
-                                    {{ node.body|markdown('comment', MATH_ENGINE, True)|reference|str|safe }}
-                                </div>
+                                    <div class="comment-body" {% if node.score <= vote_hide_threshold %}style="display:none"{% endif %}>
```
Comment (quantum5) [inline templates/comments/list.html]: nit: space inside `if`

File: templates/comments/list.html [hunk 12]
```diff
@@ -94,9 +94,19 @@ <h2><i style="padding-right: 0.3em" class="fa fa-comments"></i>{{ _('Comments')
...
+                                        {{ node.body|markdown('comment', MATH_ENGINE, True)|reference|str|safe }}
+                                    </div>
+                                    {% if node.score <= vote_hide_threshold %}
```
Comment (quantum5) [inline templates/comments/list.html]: I thought this was supposed to go first? Or does it not actually matter?

Comment (kiritofeng) [reply templates/comments/list.html]: Order doesn't matter there, but I messed up the order of https://github.com/DMOJ/site/blob/hiding_comments/templates/comments/media-js.html#L200 and https://github.com/DMOJ/site/blob/hiding_comments/templates/comments/media-js.html#L201

File: templates/comments/list.html [hunk 13]
```diff
@@ -8,7 +8,7 @@ <h2><i style="padding-right: 0.3em" class="fa fa-comments"></i>{{ _('Comments')
                 <li id="comment-{{ node.id }}" data-revision="{{ node.revisions - 1 }}"
                     data-max-revision="{{ node.revisions - 1 }}"
                     data-revision-ajax="{{ url('comment_revision_ajax', node.id) }}" class="comment">
-                    <div class="comment-display">
+                    <div class="comment-display {% if node.score <= vote_hide_threshold %}bad-comment{% endif %}">
```
Comment (quantum5) [inline templates/comments/list.html]: Space _inside_ `if`...

**Pull request merged**